pkgdb/src/pkgdb/write.cc

```diff
@@ -60,18 +59,24 @@ RulesTreeNode::addRule( AttrPathGlob & relPath, ScrapeRule rule )
                                + this->attrName + "' with new rule '"
                                + scrapeRuleToString( rule ) + "'" );
         }
-      traceLog( "assigning rule to `" + scrapeRuleToString( rule ) + "' to `"
+      traceLog( "assigning rule to '" + scrapeRuleToString( rule ) + "' to '"
                 + this->attrName + '\'' );
       this->rule = rule;
       return;
     }
-  traceLog( "adding rule to `" + this->attrName + "': `"
+  traceLog( "adding rule to '" + this->attrName + "': '"
             + displayableGlobbedPath( relPath ) + " = "
             + scrapeRuleToString( rule ) + '\'' );
 
   /* Handle system glob by splitting into 4 recursive calls. */
   if ( ! relPath.front().has_value() )
     {
+      if ( this->attrName != "legacyPackages" )
+        {
+          throw FloxException(
+            "glob in rules (null) only allowed as child of legacyPackages" );
```

Comment:
suggestion nonblocking: this doesn't let us scrape `packages` subtrees. Since we're only scraping nixpkgs at the moment that's probably fine. And it allows `["legacyPackages", "legacyPackages", null, "darwin"]`. Idk if either or those need to addressed now, but it's probably worth adding a TODO

Reply:
Good points.  I'll add that in the next related PR.

Reply:
Added to #1113